pkg/dt/fdt_linux.go

```diff
@@ -71,6 +80,36 @@ type ReserveEntry struct {
 	Size    uint64
 }
 
+// LoadFDT loads a flattened device tree from current running system.
+//
+// It first tries to load it from given file path, then
+// from /sys/firmware/fdt. If still failed, it resorts
+// to read /proc/device-tree, and flatten it.
+func LoadFDT(dtbPath string) (*FDT, error) {
+	fdtFile, err := os.Open(dtbPath)
+	if err == nil {
+		defer fdtFile.Close()
+		fdt, err := ReadFDT(fdtFile)
+		if err == nil {
+			return fdt, nil
+		}
+	}
+
+	// Fallback to load fdt from sysfs.
+	sysFDTFile, err := os.Open(sysfsFDT)
+	if err == nil {
+		defer sysFDTFile.Close()
+		fdt, err := ReadFDT(sysFDTFile)
+		if err == nil {
+			return fdt, nil
+		}
+	}
+
+	return nil, errLoadFDT
+
+	// TODO(10000TB): read dt from /proc/device-tree and flatten it.
```

Comment:
I don't think you should put in the doc that it does something it does not.

Reply:
removed.